File: 2026-01/src/day_12/boj_13305/Main.java
package day_12.boj_13305;

import java.io.*;
import java.util.*;

public class Main {

    static int MAX_PRICE = 1_000_000_000;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int n = Integer.parseInt(br.readLine());

        int[] dist = new int[n - 1];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n - 1; i++) {
            dist[i] = Integer.parseInt(st.nextToken());
        }

        int[] price = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            price[i] = Integer.parseInt(st.nextToken());
        }

        long minPrice = MAX_PRICE;
        long sum = 0;
        for (int i = 0; i < n - 1; i++) {
            minPrice = Math.min(minPrice, price[i]);
            sum += minPrice * dist[i];
        }

        System.out.println(sum);
    }
}

File: 2026-01/src/day_12/boj_13305/Main.cpp
#include <bits/stdc++.h>
using namespace std;

constexpr int MAX_PRICE = 1000000000;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;

    vector<int> dist(n - 1);
    for (int& x : dist) cin >> x;

    vector<long long> price(n);
    for (long long& x : price) cin >> x;

    long long mn = MAX_PRICE;
    long long sum = 0;
    for (int i = 0; i < n - 1; i++) {
        mn = min(mn, price[i]);
        sum += mn * dist[i];
    }

    cout << sum;
}
